packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -202,16 +206,34 @@ export class GarbageCollector implements IGarbageCollector {
 				try {
 					// For newer documents, GC data should be present in the GC tree in the root of the snapshot.
 					const gcSnapshotTree = baseSnapshot.trees[gcTreeKey];
-					if (gcSnapshotTree !== undefined) {
-						return getGCDataFromSnapshot(gcSnapshotTree, readAndParseBlob);
+					if (gcSnapshotTree === undefined) {
+						// back-compat - Older documents will have the GC blobs in each data store's snapshot tree.
+						return getSnapshotDataFromOldSnapshotFormat(
+							baseSnapshot,
+							createParams.metadata,
+							readAndParseBlob,
+						);
 					}
 
-					// back-compat - Older documents will have the GC blobs in each data store's snapshot tree.
-					return getSnapshotDataFromOldSnapshotFormat(
-						baseSnapshot,
-						createParams.metadata,
-						readAndParseBlob,
-					);
+					let snapshotData: IGarbageCollectionSnapshotData | undefined =
+						await getGCDataFromSnapshot(gcSnapshotTree, readAndParseBlob);
+					// If the GC version in base snapshot does not match the GC version currently in effect, the GC data
+					// in the snapshot cannot be interpreted correctly. Set everything to undefined except for
+					// deletedNodes because irrespective of GC versions, these nodes have been deleted and cannot be
+					// brought back. The deletedNodes info is needed to identify when these nodes are used.
+					if (
+						this.configs.gcVersionInBaseSnapshot !==
+						this.summaryStateTracker.currentGCVersion
+					) {
+						snapshotData = snapshotData
+							? {
+									gcState: undefined,
+									tombstones: undefined,
+									deletedNodes: snapshotData.deletedNodes,
+							  }
+							: undefined;
+					}
```

Comment:
Alternate approach if you like it better:
```suggestion
					if (
						this.configs.gcVersionInBaseSnapshot !==
							this.summaryStateTracker.currentGCVersion &&
						snapshotData !== undefined
					) {
						snapshotData.gcState = undefined;
						snapshotData.tombstones = undefined;
					}
```